raster-foundry/raster-foundry
# Manage state in lab using Redux

## Overview
This is pretty much a lab rewrite to use redux for state management.

### Checklist

~- [ ] Styleguide updated, if necessary~
~- [ ] Swagger specification updated, if necessary~
~- [ ] Symlinks from new migrations present or corrected for any new migrations~

## Running list of issues:
- Need better handling of http requests when updating tools - if you have multiple updates close together, some of them can get ignored (try rapidly changing bands a few times)
  - Solution: batch updates together, with more recent ones overwriting changes in the previous ones. Only allow one update request at a time.

## Testing Instructions
* Create a new tool from a template.
- verify that you can't move nodes in the create view
- verify that the name you choose gets used
* On a fresh tool, verify that you can't preview the tool or change any view related settings until you've specified all the inputs
* Verify that changes are saved automatically, and everything that should refresh does refresh
* Verify that you can change the tool name, and it doesn't overwrite changes made in the main tool view
* Verify that you can preview nodes, and the view updates when you change things
* Verify that you can still use the preview tools on the map (I removed the compare-to dropdown menu on nodes for now)
* Try to break stuff

Closes #2661 
Closes #2660
